Sources/Basic/TemporaryFile.swift

```diff
@@ -0,0 +1,172 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2016 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+
+ -------------------------------------------------------------------------
+
+ This file provides API to create and use temporary files and directories.
+*/
+
+import libc
+import func POSIX.getenv
+import class Foundation.NSFileHandle
+import var Foundation.NSUTF8StringEncoding
+import func Utility.fputs // FIXME: we should move file related helper methods to Basic.
+
+public enum TempFileError: ErrorProtocol {
+    /// Could not create a unique temporary filename.
+    case couldNotCreateUniqueName
+    /// The last 6 characters of template were not XXXXXX.
+    case badTemplateName
```

Comment:
This should never happen, since the code itself controls the template, so I don't think it is worth creating a case for it.